apps/cli/src/commands/scan.command.ts

```diff
@@ -0,0 +1,179 @@
+import BaseCommand from '@/commands/base.command'
+import type {
+  CommandActionData,
+  CommandOption
+} from '@/types/command/command.types'
+import { execSync } from 'child_process'
+import { readFileSync, statSync } from 'fs'
+import { globSync } from 'glob'
+import path from 'path'
+import secretDetector from 'secret-scan'
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const colors = require('colors/safe')
+
+export default class ScanCommand extends BaseCommand {
+  getOptions(): CommandOption[] {
+    return [
+      {
+        short: '-f',
+        long: '--file <string>',
+        description: 'Scan a specific file'
+      },
+      {
+        short: '-c',
+        long: '--current-changes',
+        description:
+          'Scan only the current changed files that are not committed'
+      }
+    ]
+  }
+
+  getName(): string {
+    return 'scan'
+  }
+
+  getDescription(): string {
+    return 'Scan the project to detect any hardcoded secrets'
+  }
+
+  action({ options }: CommandActionData): Promise<void> | void {
+    const { currentChanges, file } = options
+
+    console.log('\n=============================')
+    console.log(colors.cyan.bold('🔍 Secret Scan Started'))
+    console.log('=============================')
+
+    if (file) {
+      console.log(`Scanning file: ${file}`)
+      this.scanFiles([file as string])
+      return
+    }
+    if (currentChanges) {
+      console.log('Scanning only the current changes')
+      const files = this.getChangedFiles()
+      this.scanFiles(files)
+    } else {
+      console.log('\n\n📂 Scanning all files...\n')
+      const files = this.getAllFiles()
+      this.scanFiles(files)
+    }
+  }
+
+  private scanFiles(allfiles: string[]) {
+    const foundSecrets = []
```

Comment:
A better name would be `detectedSecrets`

Reply:
Nope, because this is just scanning the file for secret, but it is not exactly detecting the secret